assets/hip-1056/protobuf/stream/output/state_changes.proto

```diff
@@ -0,0 +1,865 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+option java_package = "com.hedera.hapi.block.stream.output.protoc";
+// <<<pbj.java_package = "com.hedera.hapi.block.stream.output">>> This comment is special code for setting PBJ Compiler java package
+option java_multiple_files = true;
+
+import "google/protobuf/wrappers.proto";
+import "basic_types.proto";
+import "exchange_rate.proto";
+import "state/addressbook/node.proto";
+import "state/blockrecords/block_info.proto";
+import "state/blockrecords/running_hashes.proto";
+import "state/blockstream/block_stream_info.proto";
+import "state/congestion/congestion_level_starts.proto";
+import "state/consensus/topic.proto";
+import "state/contract/bytecode.proto";
+import "state/contract/storage_slot.proto";
+import "state/file/file.proto";
+import "state/recordcache/recordcache.proto";
+import "state/roster/roster.proto";
+import "state/roster/roster_state.proto";
+import "state/schedule/schedule.proto";
+import "state/throttles/throttle_usage_snapshots.proto";
+import "state/token/account.proto";
+import "state/token/account_pending_airdrop.proto";
+import "state/token/network_staking_rewards.proto";
+import "state/token/nft.proto";
+import "state/token/staking_node_info.proto";
+import "state/token/token.proto";
+import "state/token/token_relation.proto";
+import "state/platform_state.proto";
+import "timestamp.proto";
+
+/**
+ * A set of state changes.
+ *
+ * Each set of changes in the network deterministically mutates the
+ * current state to a new state, and all nodes MUST apply the same
+ * changes in the same order.<br/>
+ * Each change set described in the block stream SHALL describe an
+ * ordered set of mutations which mutate the previous valid state to
+ * produce a new valid state.<br/>
+ * The order of state change sets SHALL be determined by the
+ * `consensus_timestamp`, which is a strictly ascending value
+ * determined by network consensus.
+ *
+ * ### Consensus Timestamp
+ * This value enables a consumer of the block stream to order state
+ * changes by a consistent ascending value that is determined by network
+ * consensus. A primary use case would be to enter state changes in a
+ * time-series database.<br/>
+ * This value depends on the cause of the state change.
+ *  1. For transactions, this is the transaction consensus timestamp.
+ *  1. For events without transactions, this is the consensus timestamp of
+ *     the event (round?).
+ *  1. For changes that are not the result of a transaction, but still follow
+ *     a transaction within an event, this is the consensus timestamp of the
+ *     preceding transaction.
+ */
+message StateChanges {
+    /**
+     * The consensus timestamp of this set of changes.
+     * <p>
+     * This value SHALL be deterministic with the cause of the state change.
+     */
+    proto.Timestamp consensus_timestamp = 1;
```

Comment:
Is this timestamp necessary? Won't it be the same as the preceding TransactionResult?